### DISCUSSION TITLE
Changing player starting faction

### DISCUSSION THREAD

**Discussion by XDSenDX** [Vdauphin/HeartsAndMinds#1541] [Question and answer] [answered]
Hello is it possible to change players' faction? I tried dropping in a soldier from another faction but it eventually loaded the default one.

Cheers!
PS: I deleted the default officer and replaced with officer from USMC, pasted the function in but it's not working.

**Comment by Vdauphin** [comment, marked as answer]
Hello,

Here is the custom loadout loaded: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/define_mod.sqf#L51
You can also disable this feature https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L434-L439

Cheers

**Reply by lawayne** [reply]
Hey there. I see how I can change the pre-set loadouts or disable them, but how can I also restrict/whitelist the arsenal? It would be helpful if just having the proper mods loaded automatically loads a faction preset rather than having to extensively modify the mission files.

**Reply by Vdauphin** [reply]
Hello,

> but how can I also restrict/whitelist the arsenal?

https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/define_mod.sqf#L21-L46
Also https://ace3.acemod.org/wiki/framework/arsenal-framework

> It would be helpful if just having the proper mods loaded automatically loads a faction preset rather than having to extensively modify the mission files.

PR welcome if you know how to do this

Cheers

**Reply by lawayne** [reply]
Thanks for pointing that out, I must have missed that!

As for how to adapt the mission content based on mods loaded, I know that Antistasi has (or had in previous versions) a system in place to do this. Otherwise, their new GUI now lets you select factions to fight as/with. 

KP Liberation has a system that does not automatically detect, but lets the host choose from preset factions and arsenal whitelists based on their preference.

I'd love to be able to give you more guidance on how exactly to go about implementing the system, but my mission scripting skills are just not there yet. Hopefully this gives you an idea of how you might proceed should you choose to, though.
